src/odemis/driver/spectrometer.py

```diff
@@ -262,6 +262,11 @@ def stop_generate(self):
         logging.debug("Spectrometer acquisition finished")
 
     def synchronizedOn(self, event):
+        # Trick: for the frameDuration VA, all the settings must be applied... but this normally
+        # only happens when starting the acquisition. So we force the settings to be applied when
+        # setting synchronization.
+        if event is not None:
```

Comment:
```suggestion
        if event.GetEventObject():
```
Maybe?

Reply:
This is not a wx.Event, but a odemis.model.Event *or None*. If passed `None`, it means that the synchronization is disabled. 
I'll add the type to the function signature to clarify.